pkg/client/cache/listers.go

```diff
@@ -69,10 +69,17 @@ func (s *StoreToNodeLister) List() (machines api.NodeList, err error) {
 // rather than a method of StoreToNodeLister.
 // GetNodeInfo returns cached data for the minion 'id'.
 func (s *StoreToNodeLister) GetNodeInfo(id string) (*api.Node, error) {
-	if minion, ok := s.Get(id); ok {
-		return minion.(*api.Node), nil
+	minion, exists, err := s.Get(&api.Node{ObjectMeta: api.ObjectMeta{Name: id}})
```

Comment:
Correction: If Node is namespaced uniformly, it could be hard-coded here. If Node is namespaced ad-hoc, the namespace or Node would have to become an arg of the interfaces to provide cache query context.